app/models/revision_reason.rb

```diff
@@ -9,9 +9,18 @@ class RevisionReason < ApplicationRecord
 
   before_validation :normalize_reason_code
 
+  # Callback to discard the record if marked for discard
+  before_save :discard_if_marked
+
   private
 
   def normalize_reason_code
     self.reason_code = reason_code.parameterize(separator: "_") if reason_code.present?
   end
+
+  def discard_if_marked
+    return if discarded_at.present?
```

Comment:
check if there was a change to discard_at if if discarded was initially nil. Other wise this might update the discarded_at to a new value when it wasn't intended to

Reply:
good catch

Reply:
actually, I believe this is already prevented by the return if discarded_at.present? guard clause
this function will never update discarded_at if it is present.

Reply:
okk I misread. Should be fine